File: ios/src/api/googleSheets.js
import axios from 'axios';

// Fetch standings from Google Sheets
export async function getStandings(spreadsheetId, sheetName = 's4-standings', apiKey) {
  const url = `https://sheets.googleapis.com/v4/spreadsheets/${spreadsheetId}/values/${sheetName}?key=${apiKey}`;
  const res = await axios.get(url);
  return res.data.values;
}

File: ios/API/GoogleSheets.swift
import Foundation

func fetchStandings(spreadsheetId: String, sheetName: String = "s4-standings", apiKey: String) async throws -> [[String]] {
    let urlString = "https://sheets.googleapis.com/v4/spreadsheets/\(spreadsheetId)/values/\(sheetName)?key=\(apiKey)"
    guard let url = URL(string: urlString) else {
        throw URLError(.badURL)
    }
    let (data, _) = try await URLSession.shared.data(from: url)
    let response = try JSONDecoder().decode(SheetResponse.self, from: data)
    return response.values
}

private struct SheetResponse: Decodable {
    let values: [[String]]
}
